src/tests/test_generic_server.py

```diff
@@ -0,0 +1,165 @@
+        self.assertTrue(isinstance(self.svr, generic_client.GenericClient))
+        self.assertTrue(isinstance(self.svr, generic_server.GenericServer))
+
+    def test_handle_generic_commands(self):
+        gsi = generic_server.GenericServerInterface()
+        with mock.patch('Trace.do_print') as do_print:
+            with mock.patch('Trace.dont_print')as dont_print:
+                with mock.patch('Trace.do_log') as do_log:
+                    with mock.patch('Trace.dont_log') as dont_log:
+                        with mock.patch('Trace.do_alarm') as do_alarm:
+                            with mock.patch('Trace.dont_alarm') as dont_alarm:
+                                self.svr.handle_generic_commands(gsi)
+                                self.assertFalse(do_print.called)
+                                self.assertFalse(dont_print.called)
+                                self.assertFalse(do_log.called)
+                                self.assertFalse(dont_log.called)
+                                self.assertFalse(do_alarm.called)
+                                self.assertFalse(dont_alarm.called)
+                                gsi.do_print.append('test')
+                                gsi.dont_print.append('test')
+                                gsi.do_log.append('test')
+                                gsi.dont_log.append('test')
+                                gsi.do_alarm.append('test')
+                                gsi.dont_alarm.append('test')
+                                self.svr.handle_generic_commands(gsi)
+                                self.assertTrue(do_print.called)
+                                self.assertTrue(dont_print.called)
+                                self.assertTrue(do_log.called)
+                                self.assertTrue(dont_log.called)
+                                self.assertTrue(do_alarm.called)
+                                self.assertTrue(dont_alarm.called)
+
+    def test_get_log_name(self):
+        logn = self.svr.get_log_name('foo.bar')
+        self.assertEqual(logn, 'FOO.BAR')
+        logn = self.svr.get_log_name('foo.bar.baz')
+        self.assertEqual(logn, 'FOO.BAR.BAZ')
+
+    def test_get_name(self):
+        nm = self.svr.get_name('foo')
+        self.assertEqual(nm, 'FOO', nm)
+
+    def test_server_bind(self):
+        socket = mock.MagicMock()
+        setsockopt = mock.MagicMock()
+        bind = mock.MagicMock()
+        socket.setsockopt = setsockopt
+        socket.bind = bind
+        self.svr.socket = socket
+        self.svr.server_bind()
+        self.assertTrue(setsockopt.called)
+        self.assertTrue(bind.called)
+
+    def test_serve_forever_error(self):
+        with mock.patch('Trace.alarm') as alarm:
+            self.svr.serve_forever_error('whoopsie')
+            alarm.assert_called_with(
+                e_errors.ALARM,
+                'Exception in file ??? at line -1: (None, None).  See system log for details.')
```

Comment:
I would prefer to use mock.ANY here instead of the specific message text. I think it makes sense to ensure an alarm is fired but having to keep the unit test's message text exactly right is a bit, sort of like over-fitting. WDYT?

Reply:
your suggestion is much more robust.  Done